src/pages/api/datasets.ts

```diff
@@ -0,0 +1,14 @@
+import type {NextApiRequest, NextApiResponse} from 'next';
+import {DatasetFetcher} from '@/lib/dataset-fetcher';
+
+const datasetFetcher = new DatasetFetcher({
+  endpointUrl: process.env.SEARCH_PLATFORM_ELASTIC_ENDPOINT_URL as string,
+});
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const searchResults = await datasetFetcher.search();
```

Comment:
Minor: `searchResult` instead of `searchResults` (plural), to be consistent with the DatasetFetcher's type, `SearchResult`?